packages/amplify-graphql-auth-transformer/src/utils/schema.ts

```diff
@@ -345,32 +345,35 @@ export const getSubscriptionFieldNames = (
     typeName: string;
   }> = new Set();
 
-  if (modelDirectiveConfig?.subscriptions?.level === SubscriptionLevel.on) {
-    if (modelDirectiveConfig?.subscriptions?.onCreate && modelDirectiveConfig.mutations?.create) {
-      for (const fieldName of modelDirectiveConfig.subscriptions.onCreate) {
-        fields.add({
-          typeName: 'Subscription',
-          fieldName,
-        });
-      }
+  const subscriptionLevel = modelDirectiveConfig?.subscriptions?.level ?? SubscriptionLevel.on;
+  if (subscriptionLevel !== SubscriptionLevel.on) {
+    return fields;
+  }
+
+  if (modelDirectiveConfig?.subscriptions?.onCreate && modelDirectiveConfig.mutations?.create) {
+    for (const fieldName of modelDirectiveConfig.subscriptions.onCreate) {
+      fields.add({
```

Comment:
I see these fields are just used within the Auth transformer and we are not mutating any final compiled schema. Is that taken care of in the model transformer where we add these custom names in the compiled schema? An e2e test would be nice to have to verify we handle custom sub field names correctly, if we don't have it already.

Reply:
This block doesn't look like it is changing from the current functionality. If so it should be fine to get the fix out without adding the tests.

Reply:
This is no change in behavior here, just added an IF block to early return and avoid nesting IF blocks. The newly added unit tests confirm the resolvers are assigned to the custom operation name based on the file name. The existing tests makes sure that there is no regression.